src/secrets.js

```diff
@@ -46,16 +46,22 @@ async function getSecrets(secretRequests, client) {
                 throw error
             }
         }
-        if (!selector.match(/.*[\.].*/)) {
-            selector = '"' + selector + '"'
-        }
-        selector = "data." + selector
-        body = JSON.parse(body)
-        if (body.data["data"] != undefined) {
+
+        let value;
```

Comment:
I believe we will need to pull out [this](https://github.com/hashicorp/vault-action/pull/247/files#diff-47be61e4446c15650db3c077e509c50ccfa2b4626c65fe43bad550fe25e611a9L53) JSON parse and move it here